lib/ruby_lsp/requests/support/rubocop_diagnostic.rb

```diff
@@ -19,30 +19,21 @@ class RuboCopDiagnostic
           T::Hash[Symbol, Integer],
         )
 
-        sig { params(offense: RuboCop::Cop::Offense, uri: URI::Generic).void }
-        def initialize(offense, uri)
+        sig { params(document: Document, offense: RuboCop::Cop::Offense, uri: URI::Generic).void }
+        def initialize(document, offense, uri)
+          @document = document
```

Comment:
Can we receive just the encoding here for now? We're going to need to do some refactors on this, but I'd prefer if the diagnostics didn't have a reference to the document.